### ISSUE TITLE
[JAVA] java.lang.ExceptionInInitializerError when running JUnit tests using new CureString("string")

### ISSUE THREAD

**Issue by prouzpe3** [null8626/decancer#26]
### Description
When running JUnit tests that use this library - calling `new CuredString("value")` results in an exception
```
Caused by: java.lang.RuntimeException: [x86_64-unknown-linux-gnu] this operating system (Linux) and/or architecture (x64) is not supported.
original error:
no decancer-x86_64-unknown-linux-gnu in java.library.path: /usr/java/packages/lib:/usr/lib64:/lib64:/lib:/usr/lib
	at com.github.null8626.decancer.CuredString.<clinit>(CuredString.java:70)
	... 48 more
```
I have identified the issue is with the exception for loading the library for JUnit tests.
```java
if (CuredString.isJUnit()) {
    System.loadLibrary("decancer-" + rustTarget);
}
```
I assume it was supposed to detect test runs for regression testing of this library. But it does not distinguish it from running it from another project. It goes through the thread stack trace and checks the class name, looking for the first appearence of org.junit `element.getClassName().startsWith("org.junit.")`

If I remove the condition for JUnit tests and use the NativeUtils to load the library as is used for the production use. It works well in the unit tests as well.

_Would it be possible to create a more sophisticated approach to detecting which unit tests are run?_

### Steps to Reproduce:

1. Create a new project
2. import the library
```xml
<dependencyManagement>
    <dependencies>
        <dependency>
            <groupId>com.github.null8626</groupId>
            <artifactId>decancer</artifactId>
            <version>v3.2.0</version>
        </dependency>
    </dependencies>
</dependencyManagement>          
<repositories>
    <repository>
        <id>jitpack.io</id>
        <url>https://jitpack.io</url>
    </repository>
</repositories>
```
3. Create and run JUnit tests using this library

### Environment:

OS: Ubuntu 24
Java Version: Java 21
Library Version: v3.2.0

**Comment by null8626** [comment]
Hai! Thanks for reporting this bug. This bug should be fixed in https://github.com/null8626/decancer/commit/3fdf8e735c7c13fc217a4813773dbfa180824a0c.

Sadly, i can't publish the latest results to jitpack because the repository's history went over 500 MB. Currently i am thinking of migrating from jitpack to maven central.

**Comment by prouzpe3** [comment]
I've tried it on top of current HEAD (2a1030fad182540e4d3d025c1cc173b43f74947f) and unfortunately it finished with the same error.

**Comment by null8626** [comment]
Huh, alright

**Comment by null8626** [comment]
@prouzpe3 I think https://github.com/null8626/decancer/commit/c4ec664ebea67424c8c9457a8d307a55e6f7900f should suffice. Can you try it again on top of https://github.com/null8626/decancer/commit/ffa31ed9d561b850959edcb74dd9aab77e5e4941?

**Comment by prouzpe3** [comment]
I can confirm it works on top of ffa31ed, Great work!

**Comment by null8626** [comment]
Thank you! :heart:

**Comment by null8626** [comment]
As this bug is fixed, the issue may be closed. Though the fix will officially be released in `v3.2.1`, which is sadly delayed because of #27.
